fast/assets/templates/workflow-github.yaml

```diff
@@ -46,60 +46,82 @@ jobs:
         uses: actions/checkout@v3
 
       # set up SSH key authentication to the modules repository
+
       - id: ssh-config
         name: Configure SSH authentication
         run: |
           ssh-agent -a "$SSH_AUTH_SOCK" > /dev/null
           ssh-add - <<< "$${{ secrets.CICD_MODULES_KEY }}"
 
-      # set up authentication via Workload identity Federation
+      # set up step variables for plan / apply
+
+      - id: vars-plan
+        if: github.event.pull_request.merged != true && success()
+        name: Set up plan variables
+        run: |
+          echo "plan_opts=-lock=false" >> "$GITHUB_ENV"
+          echo "provider_file=$${{env.TF_PROVIDERS_FILE_PLAN}}" >> "$GITHUB_ENV"
+          echo "service_account=$${{env.FAST_SERVICE_ACCOUNT_PLAN}}" >> "$GITHUB_ENV"
+
+      - id: vars-apply
+        if: github.event.pull_request.merged == true && success()
+        name: Set up apply variables
+        run: |
+          echo "provider_file=$${{env.TF_PROVIDERS_FILE}}" >> "$GITHUB_ENV"
+          echo "service_account=$${{env.FAST_SERVICE_ACCOUNT}}" >> "$GITHUB_ENV"
+
+      # set up authentication via Workload identity Federation and gcloud
+
       - id: gcp-auth
         name: Authenticate to Google Cloud
-        uses: google-github-actions/auth@v0
+        uses: google-github-actions/auth@v2
         with:
-          workload_identity_provider: $${{ env.FAST_WIF_PROVIDER }}
-          service_account: $${{ env.FAST_SERVICE_ACCOUNT }}
-          access_token_lifetime: 3600s
+          workload_identity_provider: $${{env.FAST_WIF_PROVIDER}}
+          service_account: $${{env.service_account}}
+          access_token_lifetime: 900s
```

Comment:
nit: 15 minutes sounds a bit on the low end for larger stages

Reply:
well, keep in mind this is fairly easy to exfiltrate, I would err on the side of caution :)